cmd/go-cache-plugin/setup.go

```diff
@@ -62,10 +62,19 @@ func initCacheServer(env *command.Env) (*gocache.Server, *s3util.Client, error)
 
 	vprintf("local cache directory: %s", flags.CacheDir)
 	vprintf("S3 cache bucket %q (%s)", flags.S3Bucket, region)
-	client := &s3util.Client{
-		Client: s3.NewFromConfig(cfg, func(o *s3.Options) {
+	s3opts := []func(*s3.Options){
+		func(o *s3.Options) {
 			o.UsePathStyle = flags.S3PathStyle
-		}),
+		},
+	}
+	if flags.S3URL != "" {
+		vprintf("S3 URL: %s", flags.S3URL)
+		s3opts = append(s3opts, func(o *s3.Options) {
+			o.BaseEndpoint = &flags.S3URL
```

Comment:
I suggest making this
```suggestion
			o.BaseEndpoint = aws.String(flags.S3URL)
```

The meaning is the same, but makes the intent clearer.

Reply:
Oh, actually maybe a better way to do this is to add `config.WithBaseEndpoint` to the LoadDefaultConfig call, instead of here?